todo-app/src/Components/todoItem.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from "react";
+import { inject, observer } from "mobx-react";
+
+import "./styles/todoItem.css";
+import Button from "./Button";
+import NewTodoFormToggle from "./TodoItem-newTodoFormToggle";
+import EditMode from "./TodoItem-editMode";
+
+@inject("todoStore")
+@observer
+export default class TodoItem extends Component {
+  state = {
+    toBeUpdatedItem: "",
+    toBeUpdatedDeadline: ""
+  };
+
+  handleToggleEdit = () => {
+    this.props.todoStore.handleToggleEditOnItem(this.props.data.id);
+    this.setState({
+      toBeUpdatedItem: this.props.data.text,
+      toBeUpdatedDeadline: this.props.data.deadline
+    });
+  }; // good
+
+  handleCancelEdit = () => {
+    this.props.todoStore.handleToggleEditOnItem(null, true);
+  }; // good
+
+  handleUpdate = () => {
+    const { toBeUpdatedItem, toBeUpdatedDeadline } = this.state;
+    const id = this.props.data.id;
+    this.props.todoStore.handleUpdateItem(
+      id,
+      toBeUpdatedItem,
+      toBeUpdatedDeadline
+    );
+    this.handleCancelEdit();
+  };
+
+  handleRemove = () => {
+    this.props.todoStore.handleRemove(this.props.data.id);
+  }; //good
+
+  handleChangeItem = updatedItem => {
+    this.setState({ toBeUpdatedItem: updatedItem });
+  }; // good
+
+  handleChangeDeadline = updatedDeadline => {
+    this.setState({ toBeUpdatedDeadline: updatedDeadline });
+  }; // good
+
+  handleToggleDone = id => {
+    this.props.todoStore.handleToggleDone(id);
+  };
+
+  render() {
+    console.log("item here");
+    console.log(this.props.todoStore.isEditing);
+    const done = this.props.data.done;
+    return (
+      <li className="todoItem">
+        <NewTodoFormToggle
+          handleChangeDeadline={this.handleChangeDeadline}
+          handleToggleEdit={this.handleToggleEdit}
+          handleChangeItem={this.handleChangeItem}
+          data={this.props.data}
+          done={done}
```

Comment:
Thanks for checking it out @isaachinman

Reply:
Okay, I understand your question now. As I was saying via PM in Slack, this is by design within `mobx-react`.

With experience you will get very good at writing Google searches that return GitHub issues... I Googled:

> mobx-react child not rerendering

And found [this issue](https://github.com/mobxjs/mobx/issues/101) within the `mobx-react` repo.

I am also new to MobX, so this is new territory for me as well. Feel free to read the entire thread, but [this comment](https://github.com/mobxjs/mobx/issues/101#issuecomment-189818379) and the maintainer's response immediately after are the most important.

I refactored your app so that `TodoItem.js` only passes `this.props.data` to `NewTodoFormToggle.js`, and then simply added a `@observer` decorator to `NewTodoFormToggle`, and it works fine.

Sorry for the confusion, this functionality is not at all intuitive for me either, but is apparently what you sign up for with MobX.